src/odemis/gui/comp/overlay/world.py

```diff
@@ -390,18 +391,26 @@ def on_roa(self, roa):
         roi (tuple of 4 floats): left, top, right, bottom position relative to the SEM image
 
         """
-        phys_rect = self.convert_roi_ratio_to_phys(roa)
+
+        if self._scanner:
+            phys_rect = self.convert_roi_phys_to_ratio(roa)
+        else:
+            phys_rect = roa
+
+
         self.set_physical_sel(phys_rect)
         wx.CallAfter(self.cnvs.request_drawing_update)
 
     def on_left_up(self, evt):
         WorldSelectOverlay.on_left_up(self, evt)
-
         if self._roa:
             if self.active:
                 if self.get_size() != (None, None):
                     phys_rect = self.get_physical_sel()
-                    rel_rect = self.convert_roi_phys_to_ratio(phys_rect)
+                    if self._scanner:
+                        rel_rect = self.convert_roi_phys_to_ratio(phys_rect)
+                    else:
+                        rel_rect = phys_rect
```

Comment:
Actually, it's a little confusing to have the variable still named `rel_rect` (even if it contains physical coordinates). So I'd suggest to rename it to `rect` or `roa`.